lib/watson-discovery-setup.js

```diff
@@ -75,44 +80,46 @@ WatsonDiscoverySetup.prototype.findDiscoveryEnvironment = function(params) {
         console.error(err);
         return reject(new Error('Failed to get Discovery environments.'));
       } else {
-        const environments = data.environments;
+        const environments = data.result.environments;
         // If a DISCOVERY_ENVIRONMENT_ID is set, validate it and use it (or fail).
         const validateID = process.env.DISCOVERY_ENVIRONMENT_ID;
         // Otherwise, look (by name) for one that we already created.
         const DISCOVERY_ENVIRONMENT_NAME = process.env.DISCOVERY_ENVIRONMENT_NAME || params.default_name;
         // Otherwise we'll reuse an existing environment, if we find a usable one.
-        let reuseEnv;
+        let reuseEnv = true;
 
+        console.log('environments[0]: ' + JSON.stringify(environments[0], null, 2));
         for (let i = 0, size = environments.length; i < size; i++) {
           const environment = environments[i];
+          console.log('environment: ' + JSON.stringify(environment, null, 2));
           if (validateID) {
             if (validateID === environment.environment_id) {
               console.log('Found Discovery environment using DISCOVERY_ENVIRONMENT_ID.');
               console.log(environment);
-              params.environment_id = environment.environment_id;
+              params.environmentId = environment.environment_id;
               return resolve(params);
             }
           } else {
             if (environment.name === DISCOVERY_ENVIRONMENT_NAME) {
               console.log('Found Discovery environment by name.');
               console.log(environment);
-              params.environment_id = environment.environment_id;
+              params.environmentId = environment.environment_id;
               return resolve(params);
-            } else if (!environment.read_only) {
-              reuseEnv = environment;
+            } else if (environment.read_only) {
+              reuseEnv = false;
             }
           }
         }
         if (validateID) {
           return reject(new Error('Configured DISCOVERY_ENVIRONMENT_ID=' + validateID + ' not found.'));
         } else if (reuseEnv) {
           console.log('Found existing Discovery environment to use: ', reuseEnv);
-          params.environment_id = reuseEnv.environment_id;
+          params.environmentId = reuseEnv.environment_id;
```

Comment:
The above change broke this.  For some reason the reuseEnv var is now being treated as a boolean and this code won't work anymore.  Probably just undo the above change.  Definitely do not leave this code trying to evaluate true.environment_id

Reply:
It used to be undefined or a found, not-read-only environment to try to use.

Reply:
ok - thought this was an issue because tests failed on this. Will take another look